framework/src/media/MediaRecorderImpl.cpp

```diff
@@ -199,11 +203,13 @@ void MediaRecorderImpl::unprepareRecorder(recorder_result_t& ret)
 
 	if (mCurState == RECORDER_STATE_NONE || mCurState == RECORDER_STATE_IDLE) {
```

Comment:
Should we handle state `PLAYER_STATE_PLAYING` and `RECORDER_STATE_PAUSED` ?
Think about below two cases:
- App misuse API :  call `unprepare()` when recorder is playing.
- App call `pause()` and then `unprepare()`, `destroy()`.   

Currently, `mrw.setCurrentRecorder(nullptr)` is called only when `stopRecorder()`, 
so in above two cases, `RecorderWorker` still keep a `shared_ptr<MediaRecorderImpl>`, 
that means `MediaRecorderImpl` instance is not deleted even if `MediaRecorder` has been destroyed.

Reply:
Unprepare can not be called during play. The state is not correct and returns an error.

Reply:
if (mCurState == RECORDER_STATE_NONE || mCurState == RECORDER_STATE_IDLE)

Reply:
A mediarecorder object has a single pair of impls. There is no reason to release unless the mediarecorder disappears from the stack.

Reply:
- Where is this error handling?
- How about case 2, `pause()` and then `unprepare()` ?

Reply:
I mean both MediaRecorder and RecorderWorker hold a shared_ptr<MediaRecorderImpl>, even MediaRecorder disappeared from stack, but static instance of RecorderWorker may still hold the Impl ( if stopRecoder() was not called ).

Reply:
@zhouxinhe 
pause --> pauseRecorder ( this is async call)
unprepare --> unprepareRecorder (this is sync call)
Each of the function name + recorder handles the function of the worker.

Reply:
i understood your means.
MediaRecorder deconstructor should handling about stop and unprepare. right?

Reply:
@zhouxinhe i think, it is user problem. 
for example)
user do malloc the value
after don't free the value.